packages/api/models/base.js

```diff
@@ -468,12 +467,15 @@ class Manager {
           return;
         }
       }
-      if (this.parseEmptyFilesArrayAsNull) {
-        // If files are [], *and* files are assume removal of record as this value
+      if (this.parseEmptyFilesArrayAsNull && fieldName === 'files') {
+        // If files are [], *and* files are to be mutated assume removal of record as this value
         // has that effect for PostGres
-        if (fieldName === 'files' && mutableFieldNames.includes(fieldName) && isEqual(value, [])) {
-          fieldsToDelete.push(`#${fieldName}`);
-          ExpressionAttributeNames[`#${fieldName}`] = fieldName;
+        if (isEqual(value, [])) {
+          if (mutableFieldNames.includes(fieldName)) {
+            fieldsToDelete.push(`#${fieldName}`);
+            ExpressionAttributeNames[`#${fieldName}`] = fieldName;
+            return;
+          }
           return;
```

Comment:
I don't quite understand the difference from previous implementation

Reply:
@jennyhliu  see the unit test updates here https://github.com/nasa/cumulus/pull/3189/files#diff-5db7829f5b5faa6a3ce55a161353fbcec8489ae540fad4fb2ada5194415c8432R1156

The issue is that files is updated ONLY if it doesn't already exist *OR* if it's in the mutable list.  

Meaning that in the prior implementation if a granule is: 

{ granuleId: someId, collectionId: someId, files: [], ....} 

and it's *not* in the mutable field names, *but* the value isn't set, whatever is in `item.files` gets written.   

In the case of values, this is intended.   In the case of null, it's handled, and in the case of it not being defined, nothing gets written as expected. 

`[]` is a special case, however as we want it to remain undefined.   This update decouples (and refactors):

`if (this.parseEmptyFilesArrayAsNull)` -- If we care about the files key being parsed as null
`(fieldName === 'files' && mutableFieldNames.includes(fieldName) && isEqual(value, []))` -- If the field name is files, *and* it's supposed to be mutated *and* it's the special value:

set the value to be deleted and add the expression attribute for the query, but don't set anything.       In this case, if the field name is not set to be mutable, it doesn't do anything.   Critically that means that *it may still write* if the value isn't present. 

Instead we say 

`if (this.parseEmptyFilesArrayAsNull && fieldName === 'files')` - If the field name matches and it's configured to handle it (refactor)
`if (isEqual(value, [])) {` -- if it's the special value, take action, if not continue as normal 
`if (mutableFieldNames.includes(fieldName))` { --  if it's a mutable field, set it to be deleted.  

 IF *not* just return, which does not add it to the list of updates, which means it won't modify it if it exists, and will not insert anything if it does not. 
 
 I'd be happy to demo through that if that doesn't make sense/feels incorrect/etc.